src/GroupAction/GroupActionCollection.php

```diff
@@ -46,17 +46,46 @@ public function addToFormContainer(Container $container): void
 		}
 
 		/**
-		 * First foreach for filling "main" select
+		 * First foreach for adding button actions
 		 */
 		foreach ($this->groupActions as $id => $action) {
-			$main_options[$id] = $action->getTitle();
+			$control = null;
+
+			if ($action instanceof GroupButtonAction) {
+				$control = $container->addSubmit((string)$id, $action->getTitle());
+				$control->onClick = $action->onClick;
+
+				/**
+				 * User may set a class to the form control
+				 */
+				$control->setAttribute('class', $action->getClass());
+
+				/**
+				 * User may set additional attribtues to the form control
+				 */
+				foreach ($action->getAttributes() as $name => $value) {
+					$control->setAttribute($name, $value);
+				}
+			}
+
 		}
 
-		$groupActionSelect = $container->addSelect('group_action', '', $main_options)
-			->setPrompt('ublaboo_datagrid.choose');
+		/**
+		 * Second foreach for filling "main" select
+		 */
+		foreach ($this->groupActions as $id => $action) {
+			if (! $action instanceof GroupButtonAction) {
+				$main_options[$id] = $action->getTitle();
+			}
+		}
+
+		if (count($main_options)) {
```

Comment:
Could you please change that to `if ($main_options !== [])`? It is faster then `count()'. Thanks

Reply:
👍